sklearn/cluster/spectral.py

```diff
@@ -307,6 +307,9 @@ class SpectralClustering(BaseEstimator, ClusterMixin):
         to be installed. It can be faster on very large, sparse problems,
         but may also lead to instabilities.
 
+    n_components : integer, optional, default is n_clusters
```

Comment:
```suggestion
    n_components : integer, optional, default=n_clusters
```

Reply:
Sure. Should I add the test as `def test_n_components` in test_spectral.py?

Reply:
Yes, please!

Reply:
To test that n_components is equivalent to n_clusters by default, I need to initialize the value of n_components in `__init__` as `self.n_components = n_clusters if n_components is None else n_components`.  

However, this would lead to Assertion error that happens at the last line `assert param_value == init_param.default, init_param.name` in `check_parameters_default_constructible` in sklearn/utils/estimator_checks.py because `param_value = n_clusters` if n_components is None, but `init_param.default = None`, and because `n_clusters != None`, this error occurs.

As thomasjpfan suggested, I "Set `self.n_components = n_components` and pass `self.n_components` directly to `spectral_clustering` without modifying any of the parameters in `__init__`. " But in this way, I cannot test `n_components` is equivalent to `n_clusters` by default, since `self.n_components = None` and `self.n_clusters = n_clusters`.

Any suggestion on this? Thank you